device/bgp/bgpsrtesegment.yaml

```diff
@@ -3,47 +3,490 @@ components:
+          type: integer
+          minimum: 0
+          maximum: 255
+          default: 0
+        srv6_sid:
+          description: >-
+            SRv6 SID.
+          type: string
+          format: ipv6
+        srv6_sid_endpoint_behavior:
+          description: >-
+            Optional SRv6 Endpoint Behavior and SID Structure.
+          $ref: '#/components/schemas/BgpSrte.SRv6SIDEndpointBehaviorAndStructure'
+    
+    BgpSrte.SegmentCTypeSubTlv:
+      x-status: under-review
+      description: >-
+        Type C: IPv4 Node Address with optional SID.
+      type: object
+      required: [ipv4_node_address]
+      properties:
+        flags:
+          description: >-
+            One octet of segment flags for V-Flag, A-Flag, S-Flag, B-Flag etc. as defined in
+            https://datatracker.ietf.org/doc/html/draft-ietf-idr-segment-routing-te-policy-13#section-2.4.4.2.12
+          type: integer
+          minimum: 0
+          maximum: 255
+          default: 0
+        sr_algorithm:
+          description: >-
+            specifying SR Algorithm when when A-Flag as defined in above flags.
+          type: integer
+          minimum: 0
+          maximum: 255
+          default: 0
+        ipv4_node_address:
           description: >-
-            TBD
+            IPv4 address representing a node.
+          type: string
+          format: ipv4
+        sr_mpls_sid:
+          description: >-
+            Optional SR-MPLS SID.
+          $ref: '#/components/schemas/BgpSrte.SrMplsSid'
+
+    BgpSrte.SegmentDTypeSubTlv:
+      description: >-
+        Type D: IPv6 Node Address with optional SID for SR MPLS.
+      type: object
+      required: [ipv6_node_address]
+      properties:
+        flags:
+          description: >-
+            One octet of segment flags for V-Flag, A-Flag, S-Flag, B-Flag etc. as defined in
+            https://datatracker.ietf.org/doc/html/draft-ietf-idr-segment-routing-te-policy-13#section-2.4.4.2.12
+          type: integer
+          minimum: 0
+          maximum: 255
+          default: 0
+        sr_algorithm:
+          description: >-
+            specifying SR Algorithm when when A-Flag as defined in above flags.
+          type: integer
+          minimum: 0
+          maximum: 255
+          default: 0
+        ipv6_node_address:
+          description: >-
+            IPv6 address representing a node.
           type: string
           format: ipv6
-          default: ::0
-        remaining_flag_bits:
+        sr_mpls_sid:
+          description: >-
+            Optional SR-MPLS SID.
+          $ref: '#/components/schemas/BgpSrte.SrMplsSid'
+
+    BgpSrte.SegmentETypeSubTlv:
+      description: >-
+        Type E: IPv4 Address and Local Interface ID with optional SID
+      type: object
+      required: [ipv4_node_address]
+      properties:
+        flags:
+          description: >-
+            One octet of segment flags for V-Flag, A-Flag, S-Flag, B-Flag etc. as defined in
```

Comment:
One octet bitmap for flags including  V-Flag, A-Flag, S-Flag, B-Flag etc. as defined in